app/src/main/java/dev/arkbuilders/arkmemo/ui/viewmodels/NotesViewModel.kt

```diff
@@ -52,12 +52,13 @@ class NotesViewModel
             }
         }
 
-        fun readAllNotes(onSuccess: (notes: List<Note>) -> Unit) {
+        fun readAllNotes(onSuccess: (notes: MutableList<Note>) -> Unit) {
             viewModelScope.launch(iODispatcher) {
                 notes.value = textNotesRepo.read() + graphicNotesRepo.read() + voiceNotesRepo.read()
                 notes.value.let {
                     withContext(Dispatchers.Main) {
-                        onSuccess(it.sortedByDescending { note -> note.resource?.modified })
+                        notes.value = it.sortedByDescending { note -> note.resource?.modified }
+                        onSuccess(notes.value.toMutableList())
```

Comment:
No need to pass `MutableList` here. We should not be passing mutable variable to other object unless specified

Reply:
Done.